wallet/src/Pos/Wallet/Web/Methods/History.hs

```diff
@@ -129,14 +137,19 @@ getHistoryLimited mCWalId mAccId mAddrId mSkip mLimit = do
             pure (cWalId', accIds')
         (Nothing, Just accId)   -> pure (aiWId accId, [accId])
     (unsortedThs, n) <- getHistory cWalId accIds mAddrId
-    let sortedTxh = sortByTime (Map.elems unsortedThs)
+
+    let !sortedTxh = forceList $ sortByTime (Map.elems unsortedThs)
+    logDebug "getHistoryLimited: sorted transactions"
+
+    logCTxs "Total last 20" $ take 20 sortedTxh
     pure (applySkipLimit sortedTxh, n)
   where
     sortByTime :: [(CTx, POSIXTime)] -> [CTx]
     sortByTime thsWTime =
         -- TODO: if we use a (lazy) heap sort here, we can get the
         -- first n values of the m sorted elements in O(m + n log m)
         map fst $ sortWith (Down . snd) thsWTime
+    forceList l = length l `seq` l
```

Comment:
I don't recall this being in `master`, why was it added?

Reply:
Well, there is the same `forceList` in `master`: https://github.com/input-output-hk/cardano-sl/blob/master/wallet/src/Pos/Wallet/Web/Methods/History.hs#L151
It was merged in this PR (to `master`): https://github.com/input-output-hk/cardano-sl/pull/2287/files#diff-1bfb54b702d6d93310adefeed510d053R150

Reply:
ha! And I was the author of the original port nevertheless 😂  I'm not a performance guru but I think this `forceList` is absolutely useless, as it would force to WHNF just the outermost function application, thus we would go from a single thunk to `(:) <thunk> <thunk>`. Anyway, not for this PR I guess..